legacy_scripts_backup/config-manager.sh

```
@@ -0,0 +1,234 @@
+    fi
+    
+    # Use nix-instantiate for more reliable parsing
+    if command -v nix-instantiate >/dev/null 2>&1; then
+        echo "📝 Reading from: $CONFIG_FILE"
+        echo ""
+        
+        # Read values using nix-instantiate for accuracy
+        get_config_value() {
+            local path="$1"
+            local description="$2"
+            local value=$(nix-instantiate --eval --strict --json "$CONFIG_FILE" -A "$path" 2>/dev/null | sed 's/"//g')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)